src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Windows/Native/Helpers.cs

```diff
@@ -85,6 +85,7 @@ public static void DecodeObject(this byte[] encoded, CryptDecodeObjectStructType
                     throw Marshal.GetLastWin32Error().ToCryptographicException();
 
                 byte* decoded = stackalloc byte[cb];
+                new Span<byte>(decoded, cb).Clear();
                 if (!Interop.crypt32.CryptDecodeObjectPointer(CertEncodingType.All, lpszStructType, encoded, encoded.Length, CryptDecodeObjectFlags.None, (byte*)decoded, ref cb))
                     throw Marshal.GetLastWin32Error().ToCryptographicException();
```

Comment:
BTW: It is odd that `receiver` callback is getting a buffer pointer without a length. It violates SDL rules about passing buffers around.

Reply:
I've modified it to pass the length as well.